lib/memoist.rb

```diff
@@ -27,9 +27,9 @@ def self.unmemoized_prefix(identifier=nil)
   end
 
   def self.escape_punctuation(string)
-    return string unless string.end_with?('?'.freeze, '!'.freeze)
+    string = string.is_a?(String) ? string.dup : string.to_s
```

Comment:
Right, I'll have to look if anything wouldn't be expecting the string to be modified in place.